test/e2e/metrics.go

```diff
@@ -0,0 +1,59 @@
+package e2e
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	topologyclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/test/e2e/utils"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+const (
+	rteContainerName = "resource-topology-exporter-container"
+)
+var _ = ginkgo.Describe("[RTE] metrics", func() {
+	var (
+		initialized         bool
+	)
+
+	f := framework.NewDefaultFramework("metrics")
+
+	ginkgo.BeforeEach(func() {
+		var err error
+
+		if !initialized {
+			_, err = topologyclientset.NewForConfig(f.ClientConfig())
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			_, err = f.ClientSet.CoreV1().Nodes().Get(context.TODO(), getNodeName(), metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			initialized = true
+		}
+	})
+
+	ginkgo.Context("With prometheus endpoint configured", func() {
+		ginkgo.It("should have some metrics exported", func() {
+			pods, err := utils.GetByRegex(f.ClientSet, "resource-topology-exporter-*")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(len(pods)).Should(gomega.BeEquivalentTo(1))
+
+			stdout, stderr, err := f.ExecWithOptions(framework.ExecOptions{
+				Command:            []string{"curl", "http://127.0.0.1:2112/metrics"},
+				Namespace:          getNamespaceName(),
+				PodName:            pods[0].Name,
+				ContainerName:      rteContainerName,
+				Stdin:              nil,
+				CaptureStdout:      true,
+				CaptureStderr:      true,
+				PreserveWhitespace: false,
+			})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%v", stderr)
+			gomega.Expect(stdout).ToNot(gomega.BeEmpty())
```

Comment:
Let's look in this output and make sure that the metrics we added are in there. Nothing fancy: `strings.Contains` (or similar) will be fine, no need for regexes.